ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -296,36 +304,40 @@ def take_step(
         take_action_text = {}
         take_action_value = {}
         take_action_outputs = {}
-        for brain_name, trainer in self.trainers.items():
-            action_info = trainer.get_action(curr_info[brain_name])
-            take_action_vector[brain_name] = action_info.action
-            take_action_memories[brain_name] = action_info.memory
-            take_action_text[brain_name] = action_info.text
-            take_action_value[brain_name] = action_info.value
-            take_action_outputs[brain_name] = action_info.outputs
+        with hierarchical_timer("get_actions"):
```

Comment:
The decorator is more concise and doesn't need the whitespace change, but this is probably better for parts of the code where we're making calls to an abstract class.